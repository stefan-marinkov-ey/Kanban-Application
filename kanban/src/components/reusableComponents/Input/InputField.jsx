import React, { useState } from "react";
import PropTypes from "prop-types";
import { InputStyleDiv } from "../../styleComponents/Container/Input_styled";

const InputField = ({
  value,
  label,
  placeholder,
  validation,
  type,
  onChange,
  name,
}) => {
  const [error, setError] = useState(false);

  return (
    <InputStyleDiv>
      {label && <label>{label}</label>}
      <input
        type={type}
        value={value}
        placeholder={placeholder}
        onChange={onChange}
        name={name}
      />
      {error && <span></span>}
    </InputStyleDiv>
  );
};

InputField.propTypes = {
  value: PropTypes.string,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  type: PropTypes.string,
  name: PropTypes.string,
  validation: PropTypes.array,
  onChange: PropTypes.func.isRequired,
};

InputField.defaultProps = {
  name: "",
  value: "",
  label: "",
  placeholder: "",
  type: "text",
  validation: [],
};

export default InputField;
