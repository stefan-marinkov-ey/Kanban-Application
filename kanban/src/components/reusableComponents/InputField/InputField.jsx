import React from "react";
import PropTypes from "prop-types";
import { StyleInputField } from "./StyleInputField.styled";

const InputField = ({
  value,
  label,
  placeholder,
  validation,
  type,
  onChange,
  name,
  onBlur,
}) => {
  return (
    <StyleInputField>
      {label && <label>{label}</label>}
      <input
        type={type}
        value={value}
        placeholder={placeholder}
        onChange={onChange}
        name={name}
        onBlur={onBlur}
      />
    </StyleInputField>
  );
};

InputField.propTypes = {
  value: PropTypes.string,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  type: PropTypes.string,
  name: PropTypes.string,
  validation: PropTypes.array,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
};

InputField.defaultProps = {
  name: "",
  value: "",
  label: "",
  placeholder: "",
  type: "text",
  validation: [],
  onChange: () => {},
  onBlur: () => {},
};

export default InputField;
