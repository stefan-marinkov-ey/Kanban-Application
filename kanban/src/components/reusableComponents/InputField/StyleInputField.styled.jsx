import styled from "styled-components";

export const StyleInputField = styled.div`
  width: 80%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: auto;
  color: ${(props) => props.theme.color};
  label {
    margin: 2% 0 1% 0;
  }
  input {
    width: 100%;
    :focus {
      outline-style: none;
    }
  }
`;
