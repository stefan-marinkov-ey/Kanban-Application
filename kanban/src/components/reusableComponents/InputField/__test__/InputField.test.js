import React from "react";
import InputField from "../InputField";
import { mount, shallow } from "enzyme";

describe("Input Component", () => {
  const wrapper = shallow(<InputField defaultValue="Hello" />);
  const input = wrapper.find("input");
  input.simulate("focus");
  it("should match snapshot", () => expect(input).toMatchSnapshot());

  it("Should set value to state when input is changed", () => {
    const wrapper = shallow(<InputField />);
    const input = wrapper.find("input");
    input.simulate("change", {
      preventDefault: jest.fn,
      target: { value: "foo" },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
