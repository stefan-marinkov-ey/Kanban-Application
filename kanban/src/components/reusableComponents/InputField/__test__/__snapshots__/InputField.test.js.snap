// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Should set value to state when input is changed 1`] = `ShallowWrapper {}`;

exports[`Input Component should match snapshot 1`] = `ShallowWrapper {}`;
